app/components/file/browse/node-file-list.component.ts

```diff
@@ -43,63 +38,73 @@ export class NodeFileListComponent implements OnInit, OnChanges {
+                                        this.filter.properties[0] : this.filter) as Property;
+            const quickSearch = filterProp && filterProp.value;
+            const loadPath = [this.folder, quickSearch].filter(x => Boolean(x)).join("/");
+            if (this.treeDisplay) {
+                return this.treeDisplay.initNodes(loadPath, true);
+            }
         }
-        let options = {};
-        const filter = this._buildFilter();
-        if (!filter.isEmpty()) {
-            options = {
-                filter: filter.toOData(),
-            };
-        }
-        this.data.updateParams({ poolId: this.poolId, nodeId: this.nodeId });
-        this.data.setOptions(options); // This clears the previous list objects
-        this.notFound = false;
-        return this.data.fetchNext(true);
-    }
-
-    @autobind()
-    public loadMore(): Observable<any> {
-        return this.data.fetchNext();
+        return Observable.of(true);
     }
 
     public get baseUrl() {
         return ["/pools", this.poolId, "nodes", this.nodeId];
     }
 
-    private _buildFilter() {
-        const filter: Property = this.filter as Property;
-        const quickSearch = filter && filter.value;
+    @autobind()
+    public loadPath(path: string, refresh: boolean = false): Observable<List<File>> {
+        if (!(path in this._fileProxyMap)) {
+            const options = buildTreeRootFilter(path);
+            const poolId = this.poolId;
+            const nodeId = this.nodeId;
+            this._fileProxyMap[path] = this.fileService.listFromComputeNode(poolId, nodeId, false, options);
+            this._fileProxyMap[path].hasMore.subscribe((hasMore) => {
+                this.moreFileMap[path] = hasMore;
+            });
+            this._fileProxyMap[path].status.subscribe((status) => {
+                this.status.next(status);
+            });
+            this._fileProxyMap[path].error.subscribe((error) => {
+                this.error.next(error);
+            });
+        }
+        let observable = refresh ? this._fileProxyMap[path].refresh() : this._fileProxyMap[path].fetchNext();
+        return observable.flatMap(() => {
+            return this._fileProxyMap[path].items.first();
+        });
+    }
 
-        const name = [this.folder, quickSearch].filter(x => Boolean(x)).join("/");
-        if (name) {
-            return FilterBuilder.prop("name").startswith(name);
-        } else {
-            return FilterBuilder.none();
+    private _initProxyMap(inputs) {
+        let poolIdInput: SimpleChange = inputs.poolId;
+        let nodeIdInput: SimpleChange = inputs.nodeId;
+        if ((poolIdInput && poolIdInput.previousValue && poolIdInput.currentValue !== poolIdInput.previousValue) ||
+            (nodeIdInput && nodeIdInput.previousValue && nodeIdInput.currentValue !== nodeIdInput.previousValue)) {
+            this.treeComponentService.treeNodes = [];
```

Comment:
make a private method to check this as this is a little confusing. 

_hasInputChanged(input: SimpleChange) {
    return input && input.previousValue && input.currentValue !== input.previousValue;
}

resulting in: 

if (this._hasInputChanged(poolIdInput) || this._hasInputChanged(nodeIdInput))

makes it a lot easier to read and see what is going on. 
it can then also be tested.